modules/standard/Collection.chpl

```diff
@@ -0,0 +1,116 @@
+/*
+  A 'Collection' is a data structure, a container for elements that provide support
+  for insert, lookup, remove, and iteration operations.

```

Comment:
I have no objections to that. I placed it in standard/ because the `List` module was also in standard/.